Close log build date, git branch and commit hash, #3970

low-batt opened the pull request:
This commit will:
- Add custom keys to Info.plist to contain build information
- Enable preprocessing of the Info.plist file in the Xcode build
- Enable use of an Info.plist header file in the Xcode build
- Add a shell script write_info_plist_header.sh that generates the Info.plist header file
- Add a new Xcode build phase to execute the script
- Add a new logBuildDetails method to AppDelegate

With these changes IINA will log a message during startup containing the build date, git branch and commit hash.

- [ ] This change has been discussed with the author.
- [x] It implements / fixes issue #3970.

---

**Description:**

low-batt committed:
Close log build date, git branch and commit hash, #3970

This commit will:
- Add custom keys to Info.plist to contain build information
- Enable preprocessing of the Info.plist file in the Xcode build
- Enable use of an Info.plist header file in the Xcode build
- Add a shell script write_info_plist_header.sh that generates the
  Info.plist header file
- Add a new Xcode build phase to execute the script
- Add a new logBuildDetails method to AppDelegate

With these changes IINA will log a message during startup containing the
build date, git branch and commit hash.

uiryuu commented on iina.xcodeproj/project.pbxproj:
```diff
@@ -2744,6 +2745,27 @@
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "iina/write_info_plist_header.sh\n";
```
Is this \n intended?

low-batt replied:
Xcode demands it. I was unable to remove it within Xcode editing the build phase. So with Xcode shutdown I removed it using Emacs. Confirmed with git diff. Brought up Xcode. Confirmed build still worked. Checked git status. No change. Xcode put the newline back in.

uiryuu replied:
Then it should be ok as long as it works

uiryuu commented on iina/Info.plist:
```diff
@@ -611,7 +617,7 @@ Released under GPLv3.</string>
 			</array>
 			<key>UTTypeDescription</key>
-			<string>Monkey's Audio Lossless Audio</string>
+			<string>Monkey&apos;s Audio Lossless Audio</string>
```
Also please revert this. I've seen Xcode automatically jumped between ' and &apos; multiple times...

low-batt replied:
Reverted.

low-batt committed:
Close log build date, git branch and commit hash, #3970

Updated to address review comments.

This commit will:
- Add custom keys to Info.plist to contain build information
- Enable preprocessing of the Info.plist file in the Xcode build
- Enable use of an Info.plist header file in the Xcode build
- Add a shell script write_info_plist_header.sh that generates the
  Info.plist header file
- Add a new Xcode build phase to execute the script
- Add a new logBuildDetails method to AppDelegate

With these changes IINA will log a message during startup containing the
build date, git branch and commit hash.
